Dockerfile

```diff
@@ -8,12 +8,7 @@ WORKDIR /build
 
 ARG FEATURES=default
 
-COPY ./nodes /build/nodes
-COPY ./pallets /build/pallets
-COPY ./runtimes /build/runtimes
-COPY ./support /build/support
-COPY ./Cargo.lock /build/Cargo.lock
-COPY ./Cargo.toml /build/Cargo.toml
```

Comment:
Neither new nor old are great. If there is any layer caching at all, `Cargo.toml` and `Cargo.lock` should be copied first, dependencies installed (those don't change often) and then the rest of the code should be copied and the project itself compiled. This way code changes don't result in rebuilding of everything.

Reply:
Thanks for the tip! We had dependency caching in our Dockerfile a while ago but experienced some issues. In the end there where some compile errors because of the dependency caching. This and the not so huge improvement by the cache lead us to remove the caching again. Last time we where hoping that [cargo-wharf](https://github.com/denzp/cargo-wharf) would solve the caching, but it looks like no development was done in 2y.